src/Facilitator/index.js

```diff
@@ -795,6 +815,12 @@ class Facilitator {
       );
       return Promise.reject(err);
     }
+    const latestAnchorInfo = await this.gateway.getLatestAnchorInfo();
```

Comment:
Proof verification still happens on latest block height, not the one passed as an argument 😱 

Check `this.getGatewayProof` and `this.getCoGatewayProof()`

Also `this.coGateway.getLatestAnchorInfo()` call is happening twice which is not required.

Reply:
> Proof verification still happens on latest block height, not the one passed as an argument 😱

Good catch 👍 
To confirm below should be the updated code and similar in confirmStakeIntent.
```
.proveGateway(
          **blockNumber**,
          proofData.accountData,
          proofData.accountProof,
          txOptions,
        )
``` 
```
return this.coGateway
            .confirmStakeIntent(
              staker,
              nonce,
              beneficiary,
              amount,
              gasPrice,
              gasLimit,
              hashLock,
              **blockNumber**,
              proofData.storageProof,
              txOptions,
            )
```

> Also this.coGateway.getLatestAnchorInfo() call is happening twice which is not required.

Do you see a major concern with this ? 
With the current implementation confirmRedeemIntent and getCoGatewayProof methods are loosely coupled and non dependent. Else we will need to update method signatures(latestAnchorInfo input) of getGatewayProof and  getCoGatewayProof 🤔

Reply:
1. 👍 
2. Making two geth call if only one is required doesn't seem correct. I think only blockNumber is required in `getGatewayProof` and `getCoGatewayProof` methods not `latestAnchorInfo input`

Reply:
I don't understand the 2nd point. Can we sync on call to discuss that 📲 ?